packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -245,4 +257,54 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
       allowMultiLogin: this.$config?.allowMultiLogin ?? true,
     };
   }
+
+  /**
+   * Called when we reconnection has been completed
+   */
+  private async handleConnectionLost(msg: ConnectionLostDetails): Promise<void> {
+    this.webSocketManager.handleConnectionLost(msg);
+
+    if (msg.isConnectionLost) {
+      // TODO: Emit an event saying connection is lost
+      this.$webex.logger.info('event=handleConnectionLost | Connection lost');
+    } else if (msg.isSocketReconnected) {
+      // TODO: Emit an event saying connection is re-estabilished
+      this.$webex.logger.info(
+        'event=handleConnectionReconnect | Connection reconnected attempting to request silent relogin'
+      );
+      // TODO: Add a check here for config (if it has property to auto silent relogin) (https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-574666)
+      await this.silentRelogin();
+    }
+  }
+
+  /**
+   * Called when we finish registration to silently handle the errors
+   */
+  private async silentRelogin(): Promise<void> {
```

Comment:
Over here, we simply reload the agent and get the required data.
If the agent was in available state and it got disconnected, we will change the state (same logic as in agent desktop).
Again, once all this is done, we need to update the config variable (maybe `isLoggedIn` or something) ... again this will be TODO as it is waiting for the other PR.